### PR TITLE
Improve documentation, add minor fixes

### PR DESCRIPTION
**Pull request by korotkevics**
* Improve Javadocs, MD files and module descriptions in POMs
* Add changes after Artem CR and winning with walk-over fix

_The below template was generated automatically. Please check in all the related check-boxes. You are welcome to add any extra remarks above this sentence._

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/play/PlayState.java [hunk 1]
```diff
@@ -30,10 +34,8 @@
   private boolean isGameWon;
 
   /**
-   * Creates PlayState instance.
-   *
    * @param communicationBus client server communication bus
-   * @param fleets           list of clients fleets
+   * @param fleets           list of clients' fleets
```
Comment (smolamarcin) [inline server/src/main/java/pl/korotkevics/ships/server/gamestates/play/PlayState.java]: Unnecessary ' ??

File: shared/src/test/java/pl/korotkevics/ships/shared/infra/communication/core/message/BaseMessageTest.java [hunk 2]
```diff
@@ -1,22 +1,75 @@
...
+  }
+  
+  private Fleet produceFleet() {
+    List<Ship> ships = Arrays.asList(Ship.ofMasts(Mast.ofIndex("5")),
+        Ship.ofMasts(Mast.ofIndex("2"), Mast.ofIndex("3"), Mast.ofIndex("1")),
+        Ship.ofMasts(Mast.ofIndex("21"), Mast.ofIndex("23"), Mast.ofIndex("24")));
+    return Fleet.ofShips(ships);
   }
+  
 }
```
Comment (smolamarcin) [inline shared/src/test/java/pl/korotkevics/ships/shared/infra/communication/core/message/BaseMessageTest.java]: Missing space at the end of file.

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/play/PlayState.java (hunk 1 above)
Comment (korotkevics) [reply server/src/main/java/pl/korotkevics/ships/server/gamestates/play/PlayState.java]: no:)

File: shared/src/test/java/pl/korotkevics/ships/shared/infra/communication/core/message/BaseMessageTest.java (hunk 2 above)
Comment (korotkevics) [reply shared/src/test/java/pl/korotkevics/ships/shared/infra/communication/core/message/BaseMessageTest.java]: I'll add it, thx:)

**Pull request merged**